devito/types/dense.py

```diff
@@ -1010,7 +994,11 @@ def __halo_setup__(self, **kwargs):
                 halo = (left_points, right_points)
             else:
                 raise TypeError("`space_order` must be int or 3-tuple of ints")
-            return tuple(halo if i.is_Space else (0, 0) for i in self.dimensions)
+            base = [halo if i.is_Space else (0, 0) for i in self.dimensions]
+            # left-/right-staggering require extra points
+            extra = [(-i, 0) if i < 0 else (0, i) for i in self.staggered]
```

Comment:
Wondering if all passes without this, I would think maybe OOB would take care of it and avoid this but I'm not sure.

Reply:
you suggested to do it this way, I haven't thought much about it? 
If you want, you can take a look (in another branch?) as you said you're gonna change parts of staggering anyway?